nomad/structs/node_class.go

```diff
@@ -123,6 +123,8 @@ func EscapedConstraints(constraints []*Constraint) []*Constraint {
 // computed node class optimization.
 func constraintTargetEscapes(target string) bool {
 	switch {
+	case strings.HasPrefix(target, "${unique."):
+		return true
```

Comment:
I'm not actually sure there actually are values that fall into this case (the unique attributes from fingerprinting would be `attr.unique`), but it was one of the test cases.